glotaran/model/item.py

```diff
@@ -1,393 +1,686 @@
-                if prop.glotaran_property_type == float:
-                    value = float(value)
-                elif prop.glotaran_property_type == int:
-                    value = int(value)
-                setattr(item, name, value)
+def get_item_validator_issues(
+    item: Item, model: Model, parameters: Parameters | None = None
+) -> list[ItemIssue]:
+    """Get validator issues for an item.
 
-            elif not getattr(ncls, name).glotaran_allow_none and getattr(item, name) is None:
-                raise ValueError(f"Missing Property '{name}' For Item '{ncls.__name__}'")
-        return item
+    Parameters
+    ----------
+    item: Item
+        The item.
+    model: Model
+        The model.
+    parameters: Parameters | None
+        The parameters.
+
+    Returns
+    -------
+    list[ItemIssue]
+    """
+    issues = []
+    for name, validator in [
+        (attr.name, attr.metadata[META_VALIDATOR])
+        for attr in fields(item.__class__)
+        if META_VALIDATOR in attr.metadata
+    ]:
+        issues += validator(getattr(item, name), item, model, parameters)
 
-    return from_dict
+    return issues
 
 
-def _validation_factory(cls):
-    @wrap_func_as_method(cls)
-    def validate(self, model: Model, parameters: ParameterGroup | None = None) -> list[str]:
-        f"""Creates a list of parameters needed by this instance of {cls.__name__} not present in a
-        set of parameters.
+def get_item_issues(
+    *, item: Item, model: Model, parameters: Parameters | None = None
+) -> list[ItemIssue]:
+    """Get issues for an item.
 
-        Parameters
-        ----------
-        model :
-            The model to validate.
-        parameter :
-            The parameter to validate.
-        missing :
-            A list the missing will be appended to.
-        """
-        problems = []
-        for name in self._glotaran_properties:
-            prop = getattr(self.__class__, name)
-            value = getattr(self, name)
-            problems += prop.glotaran_validate(value, model, parameters)
-        for validator, need_parameter in self._glotaran_validators.items():
-            if need_parameter:
-                if parameters is not None:
-                    problems += getattr(self, validator)(model, parameters)
-            else:
-                problems += getattr(self, validator)(model)
+    Parameters
+    ----------
+    item: Item
+        The item.
+    model: Model
+        The model.
+    parameters: Parameters | None
+        The parameters.
+
+    Returns
+    -------
+    list[ItemIssue]
+    """
+    issues = get_item_model_issues(item, model)
+    issues += get_item_validator_issues(item, model, parameters)
+    if parameters is not None:
+        issues += get_item_parameter_issues(item, parameters)
+    return issues
 
-        return problems
 
-    return validate
+def strip_type_and_structure_from_attribute(attr: Attribute) -> tuple[None | list | dict, type]:
```

Comment:
This types are not used for typechecking etc. They are just used in followup functions like: 

```py
if structure is dict:
    ...
```

which subscribed generics would unnecessary complicated. In principle, I could use enums here, but I don't see a reason. Note that this function is deeply hidden from the surface for users. and developers would not benefit here.

Reply:
```python
if structure is dict:
    ...
```
This is exactly why the type is wrong since this expression can only be true if the type of `structure` is `type[dict]`.
While a type of `dict` means that it is an instance of a `dict`

Reply:
#1160